### ISSUE TITLE
Can't add frozen data plot to Graph1 when first created

### ISSUE THREAD

**Issue by ajj** [SasView/sasview#2574]
**Describe the bug**
After starting a model and producing Graph1 plot and freezing a data set, Graph1 does not appear as an option to append to.

**To Reproduce**
Open SasView
Plot a model
Freeze the model
Go to Data Explorer and select frozen dataset
"Append To" options are empty.

**Expected behavior**
"Append To" would include  Graph1

**Screenshots**
![Screenshot 2023-08-02 at 14 42 42](https://github.com/SasView/sasview/assets/871392/6b3f5615-6799-401a-9a58-49a1ce8d507c)

**SasView version (please complete the following information):**
 - Version: 5.06b1 

**Operating system (please complete the following information):**
 - OS: MacOS 13.1

**Comment by butlerpd** [comment]
I thought @piotr fixed this a while back?! or did it not make 5.0.6? Or is my memory just going south quickly?  .. . but I just tested the release version of 5.0.6 and a graph with a theory curve does not show up as an option to append to in the released version either.

**Comment by krzywon** [comment]
Testing this on the `main` branch, you are correct @butlerpd, this is working there. Looking through our history, this might be the work done in #2298 which was linked to issue #2118. The work from this was supposed to be in release v5.0.6 and was include in PR #2407. There might have been some previous work this work depended on, but maybe @rozyczko knows better?

**Comment by rozyczko** [comment]
So it works in `main` but not in the release which has the same fix? Then maybe this issue got fixed post-5.0.6 as part of another commit?
Or, there's a behaviour difference between develop and release versions. This doesn't seem likely as this functionality doesn't depend on additional modules...

**Comment by butlerpd** [comment]
Now I'm confused .... very weird... and annoying must be something about a dependency as @rozyczko suggests? I guess there is not much to be done at this point? Unless we make another bug release off of 5.0.6 I guess.

**Comment by butlerpd** [comment]
So this is STILL a problem in 6.0.0. Trying the installer made from the main branch last week (at least that is what I think downloaded) this problem in fact *does* exist in main. So whatever that fix was either does not work with the installer or got overwritten somehow.

Note that once the `append to` is blank in the Data tab, it will remain so even if real data is loaded and the frozen theory plots are deleted. However the `Append to` is NOT blank for the theory tab.

The only way I have found so far to reset the `Append to` for the Data tab is to create a new plot.  I am pushing this up again though it is probably not the highest priority for 6.0.1?

**Comment by butlerpd** [comment]
Yes I had wondered whether that was the problem in which case why should it be. However my testing suggests that no matter how many graphs are plotted or how many data sets are in the data explorer, the `Append to` combo box remains blank. So you cannot for example add a new data set, even one you load after this unfortunate event, to any of those plots... etc etc. the only way I found (but there may be others) to get the combo box to no longer be empty is to choose any data set in the data explorer and use `Create New`. At that point **all** the plots on the screen instantly populate the combo box.

**Comment by butlerpd** [comment]
I also noticed that if I resolve the issue by creating a new data plot, all plots now become available to send data to. However, the Theory tab still only shows the theory plots as options for sending to.  If instead of sending using data to create a new plot, I create a new plot using theory curves, that new plot shows up as available to send a theory curve to, but does **not** show up as available for the data to be sent to.

Basically it seems like the code has a split personality (two coders at different times with a different understanding of the functionality?) such that part of the code seems to think that data should only be added to plots with data and theory curves should only be added to plots with theory curves, while the rest of the code thinks that theory or data should be able to be added to any active plot?  Not sure that helps in finding the issue but figured it was worth documenting in case.
